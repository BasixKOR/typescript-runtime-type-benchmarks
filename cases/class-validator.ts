import 'reflect-metadata'
import { IsNegative, MinLength } from 'class-validator'
import { Type } from 'class-transformer'
import { transformAndValidate, transformAndValidateSync } from 'class-transformer-validator'
import { Data } from '../data'

type DeeplyNested = Data['deeplyNested']

class DeeplyNestedType implements DeeplyNested {
  foo!: string
  num!: number
  bool!: boolean
}

class DataType implements Data {
  number!: number

  @IsNegative()
  neg_number!: number

  max_number!: number
  string!: string

  @MinLength(100)
  long_string!: string

  boolean!:boolean

  @Type(() => DeeplyNestedType)
  deeplyNested!: DeeplyNestedType
}

export function caseClassValidatorSync(data: Data) {
  return transformAndValidateSync(DataType, data)
}

export function caseClassValidatorAsync(data: Data) {
  return transformAndValidate(DataType, data)
}