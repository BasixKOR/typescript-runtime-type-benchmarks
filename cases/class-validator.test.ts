import { caseClassValidatorSync, caseClassValidatorAsync } from './class-validator'
import { DATA } from '../data'

describe('caseClassValidatorSync', () => {
  it('should pass', () => {
    const res = caseClassValidatorSync(DATA)
    expect(res).toEqual(DATA)

    if (res) {
      expect(res.number).toEqual(DATA.number)
    }
  })

  it('should fail negative number test', () => {
    const req = Object.assign({}, DATA, { neg_number: 1 })
    expect(() => caseClassValidatorSync(req)).toThrow()
  })

  it('should fail min length test', () => {
    const req = Object.assign({}, DATA, { long_string: '' })
    expect(() => caseClassValidatorSync(req)).toThrow()
  })
})

describe('caseClassValidatorAsync', () => {
  it('should pass', async () => {
    const res = await caseClassValidatorAsync(DATA)
    expect(res).toEqual(DATA)

    if (res) {
      expect(res.number).toEqual(DATA.number)
    }
  })

  it('should fail negative number test', () => {
    const req = Object.assign({}, DATA, { neg_number: 1 })
    return expect(caseClassValidatorAsync(req)).rejects.toMatchSnapshot()
  })

  it('should fail min length test', () => {
    const req = Object.assign({}, DATA, { long_string: '' })
    return expect(caseClassValidatorAsync(req)).rejects.toMatchSnapshot()
  })
})