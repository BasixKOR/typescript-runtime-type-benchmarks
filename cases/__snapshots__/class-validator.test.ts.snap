// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`caseClassValidatorAsync should fail min length test 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "minLength": "long_string must be longer than or equal to 100 characters",
    },
    "property": "long_string",
    "target": DataType {
      "boolean": true,
      "deeplyNested": DeeplyNestedType {
        "bool": false,
        "foo": "bar",
        "num": 1,
      },
      "long_string": "",
      "max_number": 1.7976931348623157e+308,
      "neg_number": -1,
      "number": 1,
      "string": "string",
    },
    "value": "",
  },
]
`;

exports[`caseClassValidatorAsync should fail negative number test 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isNegative": "neg_number must be a negative number",
    },
    "property": "neg_number",
    "target": DataType {
      "boolean": true,
      "deeplyNested": DeeplyNestedType {
        "bool": false,
        "foo": "bar",
        "num": 1,
      },
      "long_string": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Vivendum intellegat et qui, ei denique consequuntur vix. Semper aeterno percipit ut his, sea ex utinam referrentur repudiandae. No epicuri hendrerit consetetur sit, sit dicta adipiscing ex, in facete detracto deterruisset duo. Quot populo ad qui. Sit fugit nostrum et. Ad per diam dicant interesset, lorem iusto sensibus ut sed. No dicam aperiam vis. Pri posse graeco definitiones cu, id eam populo quaestio adipiscing, usu quod malorum te. Ex nam agam veri, dicunt efficiantur ad qui, ad legere adversarium sit. Commune platonem mel id, brute adipiscing duo an. Vivendum intellegat et qui, ei denique consequuntur vix. Offendit eleifend moderatius ex vix, quem odio mazim et qui, purto expetendis cotidieque quo cu, veri persius vituperata ei nec. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.",
      "max_number": 1.7976931348623157e+308,
      "neg_number": 1,
      "number": 1,
      "string": "string",
    },
    "value": 1,
  },
]
`;
